packages/flutter/lib/src/material/ink_sparkle.dart

```diff
@@ -0,0 +1,765 @@
+  }) {
+    return InkSparkle(
+      controller: controller,
+      referenceBox: referenceBox,
+      position: position,
+      color: color,
+      containedInkWell: containedInkWell,
+      rectCallback: rectCallback,
+      borderRadius: borderRadius,
+      customBorder: customBorder,
+      radius: radius,
+      onRemoved: onRemoved,
+      textDirection: textDirection,
+    );
+  }
+}
+
+////////////////////////////////////////
+// GENERATED CODE BELOW - DO NOT EDIT //
+////////////////////////////////////////
+
+// TODO(clocksmith): Move to separate file?
+
+/// A class for managing [FragmentProgram] that includes a pre-transpiled
+/// shader program into SPIR-V.
+class FragmentShaderManager {
+  static Future<FragmentShaderManager> inkSparkle() async {
+    final manager = FragmentShaderManager._();
+    await manager.compile();
+    return manager;
+  }
+
+  FragmentShaderManager._();
+
+  Future<void> compile() async {
+    _program = await ui.FragmentProgram.compile(spirv: spirvByteBuffer);
+  }
+
+  /// Creates a shader with the original program and optional uniforms.
+  /// 
+  /// A new shader must be made whenever the uniforms are updated.
+  Shader shader({
+    Vector4? uColor,
+    Vector2? uResolution,
+    Vector2? uTouch,
+    double? uTime,
+    double? uTouchUp,
+    double? uRippleDurationMs,
+    double? uPauseInflection,
+    double? uPostPauseTimeScale,
+    double? uPixelDensity,
+  }) {
+    return _program.shader(
+      floatUniforms: Float32List.fromList([
+        ...uColor != null ? uColor.storage : [0, 0, 0, 0],
+        ...uResolution != null ? uResolution.storage : [0, 0],
+        ...uTouch != null ? uTouch.storage : [0, 0],
+        ...uTime != null ? [uTime] : [0],
+        ...uTouchUp != null ? [uTouchUp] : [0],
+        ...uRippleDurationMs != null ? [uRippleDurationMs] : [0],
+        ...uPauseInflection != null ? [uPauseInflection] : [0],
+        ...uPostPauseTimeScale != null ? [uPostPauseTimeScale] : [0],
+        ...uPixelDensity != null ? [uPixelDensity] : [0],
+      ]),
+    );
+  }
+  
+  late ui.FragmentProgram _program;
+
+  /// Direct access to the [ui.FragmentProgram] that this class manages.
+  /// 
+  /// In general, this is not needed, but may be useful for debugging or edge cases.
+  ui.FragmentProgram get program => _program;
+
+  /// Direct access to the source GLSL that was used to generate this class.
+  /// 
+  /// In general, this is not needed, but may be useful for debugging or edge cases.
+  String get glsl => _glslString;
+
+  /// Direct access to the the SPIR-V bytecode that was used to generate this class.
+  /// 
+  /// In general, this is not needed, but may be useful for debugging or edge cases.
+  ///
+  /// Words in SPIR-V are 32 bits. Every 4 elements in this list represents 1
+  /// SPIR-V word. See https://www.khronos.org/registry/SPIR-V/.
+  ByteBuffer get spirvByteBuffer => Uint8List.fromList(_spirvByteList).buffer;
+}
+
+const _glslString = '''
```

Comment:
curious if there it would be "better" to have this be in a .glsl file that is read

Reply:
AH good question! This is not actually even necessary. It was generated as part of the generator library as a default behavior, but it is not actually needed for the InkSparkle, only the SPIR-V. Will move as a comment in the generated file instead.